exercises/03-keywords_and_variables.md

```diff
@@ -1,74 +1,83 @@
 # Keywords and variables
 
-One of the most powerful things that Robot Framework can do is to provide the capability to write test cases
-using natural language. In an ideal world, it is not just people who write those tests, but also other people are interested in the automated test case results, including
-how, where, and when they are run.
+One of the most powerful things that Robot Framework can do is to provide the capability to write test
+cases using natural language. In an ideal world, it is not just people who write those tests, but also
+other people are interested in the automated test case results, including how, where, and when they are
+run.
 
 This can be made possible by using user defined keywords:
 http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#creating-user-keywords
 
 ## User defined keywords
 
-With user defined keywords we can create abstraction layer when needed.
-A general rule of thumb is that library and resource keywords are "general sounding" ones and with user keywords in test suite you can create needed abstraction layer based on the test context.
+With user defined keywords we can create abstraction layer when needed. A general rule of thumb is that
+library and resource keywords are "general sounding" ones and with user keywords in test suite you can
+create needed abstraction layer based on the test context.
 
 <details>
     <summary>From previous exercises we should have something like this in our <code>robot/login.robot</code></summary>
 
-```
+```robot
 *** Settings ***
-Library    SeleniumLibrary
+Library    Browser
 
 *** Test Cases ***
 
 Welcome Page Should Be Visible After Successful Login
-    Open Browser    http://localhost:7272   browser=ff
-    Input Text    id=username_field    demo
-    Input Password    id=password_field    mode
-    Click Element    id=login_button
-    Page Should Contain    Welcome Page
-    Location Should Be    http://localhost:7272/welcome.html
-    Title Should Be    Welcome Page
+    New Browser    headless=${FALSE}
+    New Page    http://localhost:7272
+    Fill Text    id=username_field    demo
+    Fill Secret    id=password_field    mode
+    Click    id=login_button
+    Get Text    id=header    ==    Welcome Page
```

Comment:
I don't know if this is a nitpick, but this is less future-proof than `Page should contain` was (as evidenced by you needing to change the HTML code).

If we want to retain the idea that we want to just check that the page is correct without concerning ourselves about the page structure, we can do:
```robotframework
Get Text     body    contains    Welcome Page
```

Reply:
That's actually a better solution. Should've figured it out myself.

Reply:
Suggested changes applied and committed.